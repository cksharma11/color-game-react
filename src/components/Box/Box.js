import React from "react";

export default class Box extends React.Component {
  render() {
    return (
      <div
        className="box"
        style={{ background: this.props.color }}
        id={this.props.color}
        onClick={this.props.onClick}
      />
    );
  }
}
