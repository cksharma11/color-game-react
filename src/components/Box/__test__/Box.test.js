import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Box from "../Box";

configure({ adapter: new Adapter() });

describe("Box", () => {
  const colors = ["red", "blue", "green", "yellow"];

  it("should render properly", () => {
    const props = {
      colors: colors,
    };

    const wrapper = shallow(<Box {...props} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
