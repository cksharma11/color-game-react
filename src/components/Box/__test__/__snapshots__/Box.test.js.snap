// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render properly 1`] = `"<div className=\\"box\\" style={{...}} id={[undefined]} onClick={[Function: onClick]} />"`;

exports[`Box should render properly 1`] = `"<div className=\\"box\\" style={{...}} id={[undefined]} onClick={[undefined]} />"`;
